### ISSUE TITLE
Transform to Typescript

### ISSUE THREAD

**Issue by subzero10** [honeybadger-io/honeybadger-react#274]
Let's transform our honeybadger-react code into typescript.
As a bonus, we won't have to manually maintain typescript definitions.

**Comment by JustFly1984** [comment]
I have an issue with @honeybadger-io/react and typescript in one of projects.

An error while passing a `honeybadger` Client type into `<ErrorBoundary>` component, it expects HoneyBadger

Temporary workaround I've implemented: 

```typescript
import ErrorBoundary from '@honeybadger-io/react';
import React from 'react';

import honeybadger from '../lib/honeybadger';

import type { AppProps } from 'next/app';
import type { NextPage } from 'next';
import type Honeybadger from '@honeybadger-io/js';

const App = (): JSX.Element => {
  return (
    // eslint-disable-next-line total-functions/no-unsafe-type-assertion
    <ErrorBoundary honeybadger={honeybadger as Honeybadger}>
      <>
        <Component>
      </>
    </ErrorBoundary>
  );
};
```

Full error text:

```
No overload matches this call.
  Overload 1 of 2, '(props: Props | Readonly<Props>): ErrorBoundary', gave the following error.
    Type 'Client' is not assignable to type 'Honeybadger'.
      Type 'Client' is missing the following properties from type 'Honeybadger': errorHandler, requestHandler, lambdaHandler
  Overload 2 of 2, '(props: Props, context: any): ErrorBoundary', gave the following error.
    Type 'Client' is not assignable to type 'Honeybadger'.
      Type 'Client' is not assignable to type 'Honeybadger'.ts(2769)
```

**Comment by subzero10** [comment]
Hey @JustFly1984 , can you please try installing [branch](https://github.com/honeybadger-io/honeybadger-react/tree/fix-ts-definitions) and let us know if you still experience the issue?

npm command to install a specific branch:
```
npm install@honeybadger-io/react#fix-ts-definitions --save
```
Thanks!

**Comment by joshuap** [comment]
@JustFly1984 this should now be fixed in @honeybadger-io/react@1.0.2

**Comment by JustFly1984** [comment]
Yes, it is fixed!Thank you @joshuap
